app/views/govuk_publishing_components/components/_contents_list.html.erb

```diff
@@ -32,7 +43,12 @@
     <ol class="gem-c-contents-list__list">
       <% contents.each.with_index(1) do |contents_item, position| %>
         <li class="<%= cl_helper.list_item_classes(contents_item, false) %>" <%= "aria-current=true" if contents_item[:active] %>>
-          <% link_text = format_numbers ? cl_helper.wrap_numbers_with_spans(contents_item[:text]) : contents_item[:text] %>
+          <% link_text = format_numbers ? cl_helper.wrap_numbers_with_spans(contents_item[:text]) : contents_item[:text]
+             ga4_link_data = nil
+             if ga4_tracking
+                ga4_link_data = {"index": {"index_link": position}, "index_total": contents.length}.to_json
+             end
```

Comment:
Should be able to shorten this? Also (nitpick) please add spaces to improve readability, as shown.

```Ruby
ga4_link_data = { "index": { "index_link": position }, "index_total": contents.length }.to_json if ga4_tracking
```